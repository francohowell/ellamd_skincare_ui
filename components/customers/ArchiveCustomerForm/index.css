@import "mixins";

.wrapper {
  position: relative;
  display: inline-block;
  margin-right: 20px;
}
