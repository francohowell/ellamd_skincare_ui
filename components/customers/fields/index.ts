export * from "./AddressField";
export * from "./CustomerField";
export * from "./PhotosField";
export * from "./SetPhoneField";
export * from "./StateField";
