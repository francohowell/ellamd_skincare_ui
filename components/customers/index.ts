export * from "./ArchiveCustomerForm";
export * from "./CreateCustomerForm";
export * from "./fields";
export * from "./CustomerList";
export * from "./CustomerPage";
export * from "./UpdateCustomerForm";
export * from "./WelcomeDialog";
export * from "./Inbox";
