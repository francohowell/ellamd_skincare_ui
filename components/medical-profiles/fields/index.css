@import "colors";
@import "mixins";

.columns {
  display: flex;
  margin-left: -20px;

  & > * {
    flex: 1 0 !important;
    margin-left: 20px !important;
  }

  & select {
    margin-bottom: 5px;
  }

  @media (max-width: $phone-breakpoint) {
    display: block;
    margin-left: 0;

    & > * {
      margin-left: 0 !important;
    }
  }
}

.additional-details-label {
  color: $gray-5;
  font-size: 14px;
  display: block;
  margin-bottom: 4px;
}

.date-of-birth-field .columns {
  margin-left: -10px !important;

  & > * {
    margin-left: 10px !important;

    &:nth-child(1) { flex: 1.6 0 !important; }
    &:nth-child(2) { flex: 0.8 0 !important; }
    &:nth-child(3) { flex: 1 0 !important; }
  }
}

.label-description {
  margin: 4px 0 15px;
  line-height: 1.2;
  font-size: 14px;
  color: rgba($black, 0.5);
}

.swatch {
  width: 50px;
  height: 50px;
  border-radius: 50px;
  padding: 4px;
  background: $white;
  border: 1px solid $gray-3;
  display: block;
  margin: 5px auto;

  & span {
    display: block;
    width: 100%;
    height: 100%;
    border-radius: 100%;
  }
}

.type-number {
  font-size: 12px;
  display: block;
  margin-top: 10px;
  text-transform: uppercase;
  color: $gray-3;
}

.type-name {
  display: block;
  font-size: 18px;
  margin: 8px 0;
}

.type-description {
  font-size: 14px;
  display: block;
  margin-bottom: 10px;
  color: $gray-5;
}

.skin-types {
  margin: -8px -15px -20px;

  & input {
    display: none;
  }

  & > label {
    padding: 10px 15px;
    border-radius: 3px;
    display: inline-block !important;
    width: 16.66%;
    vertical-align: top;
    cursor: pointer;
    text-align: center;
    border: 1px solid transparent;
    margin-bottom: 0 !important;

    &:hover {
      background: rgba($blue-3, 0.1);
    }
  }

  & .checked,
  & .checked:hover {
    background: rgba($blue-3, 0.25);
    border: 1px solid $blue-3;

    & .type-number {
      color: $gray-5;
    }

    & .type-description {
      color: $dark-gray-2;
    }
  }

  @media (max-width: $phone-breakpoint) {
    margin: 0;

    & label {
      width: 33%;
    }
  }
}

:global(.pt-dialog) .skin-types {
  margin: 0;

  & label {
    width: 33%;
  }
}

.label {
  margin-bottom: 5px;
}

.concerns {
  display: flex;
  align-items: stretch;
  flex-wrap: wrap;
  justify-content: space-between;
}

.concern {
  flex: 0 0 150px;
  display: inline-block;
  cursor: pointer;
  vertical-align: top;
  margin: 0 0 20px;
  text-align: center;
  border-radius: 3px;
  padding: 20px 15px;
  border: 1px solid transparent;

  &.hidden {
    visibility: hidden;
  }

  @media (max-width: $phone-breakpoint) {
    padding: 10px 15px;
    flex: 0 0 140px;
  }

  &:hover {
    background: rgba($blue-3, 0.1);
  }

  & img {
    opacity: 0.5;
    height: 75px;
    margin: 0 auto 15px;
    display: block;
  }

  &.selected {
    background: rgba($blue-3, 0.25);
    border: 1px solid $blue-3;

    & img {
      opacity: 0.9;
    }
  }
}
