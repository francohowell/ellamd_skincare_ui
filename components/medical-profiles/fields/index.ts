export * from "./MedicalProfileField";

export * from "./DateOfBirthField";
export * from "./PreferredFragranceField";
export * from "./SexField";
export * from "./SkinConcernsField";
export * from "./SkinTypeField";
export * from "./SunscreenFrequencyField";
