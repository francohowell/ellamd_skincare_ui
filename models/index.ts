export * from "./Audit";
export * from "./Condition";
export * from "./Customer";
export * from "./Diagnosis";
export * from "./Formulation";
export * from "./Id";
export * from "./Identity";
export * from "./Ingredient";
export * from "./MedicalProfile";
export * from "./Message";
export * from "./MessageGroup";
export * from "./Pharmacist";
export * from "./Photo";
export * from "./PhotoGroup";
export * from "./Physician";
export * from "./Prescription";
export * from "./PrescriptionIngredient";
export * from "./Product";
export * from "./ProductIngredient";
export * from "./Regimen";
export * from "./RegimenProduct";
export * from "./Story";
export * from "./Stripe";
export * from "./Subscription";
export * from "./Visit";

export * from "./MedicalProfile/labels";
