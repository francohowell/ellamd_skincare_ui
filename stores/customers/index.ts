export * from "./MyCustomerStore";

export * from "./MyMedicalProfileAuditStore";
export * from "./MyMessageStore";
export * from "./MyPhotoStore";
export * from "./MyStoryStore";
