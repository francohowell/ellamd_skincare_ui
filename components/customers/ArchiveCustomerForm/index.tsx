import {Button, Dialog, Intent} from "@blueprintjs/core";
import {inject, observer} from "mobx-react";
import * as React from "react";

import {RiseTransitionGroup} from "components/common";
import {Customer} from "models";
import {CustomerStore} from "stores";
import {history, Method, request, ROUTES, Status, SubmitEvent, Toaster} from "utilities";

import * as styles from "./index.css";

interface Props {
  customer: Customer;
  customerStore?: CustomerStore;
  className?: string;
}
interface State {
  isOpen: boolean;
  isLoading: boolean;
}

@inject("customerStore")
@observer
export class ArchiveCustomerForm extends React.Component<Props, State> {
  public state: State = {
    isOpen: false,
    isLoading: false,
  };

  private toggleForm = () => {
    this.setState({isOpen: !this.state.isOpen});
  };

  private handleSubmit = async (event: SubmitEvent) => {
    event.preventDefault();

    this.setState({isLoading: true});
    const response = await request(`customers/${this.props.customer.id}/archive`, Method.POST);
    this.setState({isLoading: false});

    // tslint:disable-next-line switch-default
    if (response.status === Status.Success) {
      const {customer} = response.data! as {customer: Customer};
      this.props.customerStore!.removeCustomer(customer);
      Toaster.show({message: "Customer archived.", intent: Intent.SUCCESS, iconName: "tick"});
      this.setState({isOpen: false});
      history.push(ROUTES.dashboard);
    }
  };

  private renderConfirmation() {
    return (
      <Dialog isOpen={this.state.isOpen} onClose={this.toggleForm} title="Archive customer?">
        <div className="pt-dialog-body">
          This won't delete the Customer permanently, but they will no longer appear on the
          Administrator, Physician, or Pharmacist dashboards.
        </div>
        <div className="pt-dialog-footer">
          <div className="pt-dialog-footer-actions">
            <Button loading={this.state.isLoading} text="Cancel" onClick={this.toggleForm} />
            <Button
              loading={this.state.isLoading}
              intent={Intent.PRIMARY}
              onClick={this.handleSubmit}
              text="Archive customer"
            />
          </div>
        </div>
      </Dialog>
    );
  }

  public render() {
    return (
      <div className={[styles.wrapper, this.props.className].join(" ")}>
        <Button intent={Intent.DANGER} className="pt-minimal" onClick={this.toggleForm}>
          Archive customer
        </Button>

        <RiseTransitionGroup>
          {this.renderConfirmation()}
        </RiseTransitionGroup>
      </div>
    );
  }
}
