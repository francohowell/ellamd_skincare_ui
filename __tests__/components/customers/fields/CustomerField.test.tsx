import {CustomerField} from "components/customers/fields";
import {Customer} from "models";
import {MemoryRouter} from "react-router";
import {create} from "react-test-renderer";

import * as React from "react";

test("renders correctly", () => {
  const customerData = {
    id: 287, firstName: "Dora", lastName: "Hare", email: "customer+3@ellamd.com",
    createdAt: "2017-09-27T21:49:27.921Z", updatedAt: "2017-11-06T21:49:28.688Z", lastOnboardingStep: 4,
    physician:
      {
        id: 2, firstName: "Jack", lastName: "Physician", email: "physician@ellamd.com"
      },
    subscription:
      {
        id: 123,
        status: "inexistent",
        stripe_customer_id: "cus_123",
        initial_treatment_plan_is_free: false
      },
    visits:
      [{id: 116, createdAt: "2017-11-01T21:49:28.666Z", photos: []}],
    photos: [],
    dateOfBirth: "1988-04-15",
    sex: "female",
    addressLine1: "397 Libby Street",
    addressLine2: "Apt 2",
    zipCode: "90210",
    state: "CA",
    city: "Beverly Hills",
    skinConcerns: ["acne", "sun spots"],
    skinType: 3} as any;

  const customer = new Customer(customerData);

  const component = create(
    <MemoryRouter>
      <CustomerField field="firstName" customer={customer} />
    </MemoryRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
