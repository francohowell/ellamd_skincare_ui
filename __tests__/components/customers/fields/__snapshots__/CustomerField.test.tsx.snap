// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="pt-label"
>
  <div
    className={undefined}
  >
    First name
  </div>
  <input
    className="pt-input pt-fill pt-large"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="Dora"
  />
</label>
`;
